operator/cmd/mesh/upgrade.go

```diff
@@ -199,7 +202,11 @@ func upgrade(rootArgs *rootArgs, args *upgradeArgs, l clog.Logger) (err error) {
 	// Read the current installation's profile IOP yaml to check the changed profile settings between versions.
 	currentSets := args.set
```

Comment:
why do we even need the old charts? We can fetch the existing IOP from the in cluster IstioOperator file we persist I thought?

Reply:
There is another pre-upgrade check for the current IOP and the target IOP. Looks like we're re-generating the current IOP with the profile specified with the upgrade command.
Not sure if this is necessary, @shamsher31 may know more details about this.
I would prefer to eliminate the old chart during upgrade, especially if end-users installed the current istio with customized chart.

Reply:
It would be ideal to not have to manage the currently installed manifests, so if we could get those from the existing IOP without having to pass them in or make a call to github that would be ideal. Otherwise people will have to maintain copies of all istio versions which will become a pain very quickly.